src/averageAges.js

```diff
@@ -37,7 +38,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const mothersNames = people
+    .filter(person => person.mother)
+    .map(person => '' + person.mother);
+
+  const women = people
+    .filter(person => person.sex === 'f')
+    .filter(person => !withChildren || mothersNames.includes(person.name))
+    .map(person => person.died - person.born);
+
+  return women.reduce((a, b) => a + b) / women.length;
```

Comment:
You`re using this reduce method everywhere, you can put it in a separate function and pass the necessary parameters, but it is optional